clinica/iotools/converters/ukb_to_bids/ukb_utils.py

```diff
@@ -0,0 +1,311 @@
+from os import PathLike
+from typing import BinaryIO, Iterable, List, Tuple, Union
+
+from pandas import DataFrame
+
+
+def find_clinical_data(
+    clinical_data_directory: PathLike,
+) -> Tuple[DataFrame, DataFrame, DataFrame, DataFrame, DataFrame]:
+    from pathlib import Path
+
+    import pandas as pd
+
+    from clinica.utils.stream import cprint
+
+    # Read the xls
+    try:
+        image_data_file = list(Path(clinical_data_directory).glob("*.csv"))
+    except StopIteration:
+        raise FileNotFoundError("Imaging collection file not found.")
+
+    # Assign the dfs
+    df_list = []
+    for i in range(0, len(image_data_file)):
+        df_list.append(pd.read_csv(str(image_data_file[i])))
+        df_list[i] = df_list[i].set_index(df_list[i].axes[1][0])
+        if df_list[i].index.name == "eid":
+            df_clinical = df_list[i]
```

Comment:
I'm not sure I understand this part,  we taking the last item in `df_list` with "eid" as index and assigning it to `df_clinical`?  Do we really need to maintain the list in that case?

Reply:
We set the first column as index, and we check that it is eid. That way, we know it is the file we are looking for.
It is probably a good security to have in case the users provides us. with a directory with more than one tsv

Reply:
I agree with @omar-rifai you could just loop over the files, assign the right DataFrame to df_clinical, and not maintain the list since you're not using it afterwards.

Reply:
It seems somewhat contrived if the purpose is to check that we don't have many `.csv`files.  Moreover, if we are making the hypothesis that there can possible be more than one, why are we taking the last (as opposed to the first?).

Reply:
simplified !